### ISSUE TITLE
Resetting appvm

### ISSUE THREAD

**Issue by frozentime345** [QubesOS/qubes-issues#5177]
**The problem you're addressing (if any)**
_A clear and concise description of the problem, if any, that this enhancement is intended to address._
Sometimes I create appvms for temporary storage that I later reset by deleting and recreating, and this process is somewhat tedious.

**Describe the solution you'd like**
_If you have something in mind, a clear and concise description of what you want to happen. If you don't have something in mind, indicate as much._
Maybe create a base just like a disposable vm and when you click a reset button it clones the base to overwrite the appvm.

**Where is the value to a user, and who might that user be?**
_Which users is this most likely to benefit? What user needs does this address? How might a user summarize this change or new thing?_
Those who use disposable browsers for daily browsing and like to save bookmarks temporarily without potentially comprising their disposable template. Those who use the same appvm for multiple incompatible purposes to avoid the creation of too many appvms.

Summary: A manually resetting disposable vm.

**Describe alternatives you've considered**
_A clear and concise description of any alternative solutions or features you've considered._
Just deleting and recreating the appvm using a base I clone from.

**Additional context**
_Add any other context or screenshots about the feature request here._

**Relevant [documentation](https://www.qubes-os.org/doc/) you've consulted**
_A list of links to the Qubes documentation (or other relevant software documentation) pages you have already consulted._
I looked through the disposable and disposable customization page and found nothing like it.

**Related, [non-duplicate](https://www.qubes-os.org/doc/reporting-bugs/#new-issues-should-not-be-duplicates-of-existing-issues) issues**
_A list of links to other bug reports, feature requests, or tasks in the qubes-issues tracker. Do not describe any other unreported bugs, features, or tasks here._

**Comment by andrewdavidwong** [comment]
Do static/named DisposableVMs not already cover this?

**Comment by frozentime345** [comment]
Not exactly, I am using a static one for something else I believe and it automatically resets whereas I'm looking for something I can manually reset(ie when I'm done with the bookmarks, the appvm gets compromised or I wish to use the appvm for something else). So in terms of current features it would be more like cloning a base vm to overwrite an appvm manually rather than it recreating from a base each time like a disposable.

**Comment by zS3mNH2** [comment]
To me it sounds like you want restore points / snapshots. Basically, the ability to snapshot a particular VM and then rollback to that snapshot later on. It is fundamentally the same as cloning, except cloning creates "VMs" in addition to simply snapshotting the disk that is used.

**Comment by frozentime345** [comment]
Well I was thinking more just base version and current version but that is the idea yeah. More specifically though I know it's the same as cloning but I just thought it would be nice to have a rollback/reset button to automate the overwrite process. Not a big deal, just thought it would be a cool type of vm setup.

**Comment by tasket** [comment]
@frozentime345 [Qubes-VM-hardening](https://github.com/tasket/Qubes-VM-hardening) enables this kind of _'reset everything except these whitelisted files'_ use case.

**Comment by frozentime345** [comment]
That looks like a little more setup than what I was thinking, rather than whitelisting files as I added them I was looking more to reset once in a while. Also. that seems to be at the template level for configuring things whereas I was thinking more like a base and in-use vm set so that I can do any customization needed in each base. It looks interesting but just cloning vms seems easier at this point for my use case(occasional reset).

**Comment by ben-grande** [comment]
There is `vm-config` feature now. You can also revert volumes:

- https://github.com/QubesOS/qubes-issues/issues/5881
